include/deal.II/base/utilities.h

```diff
@@ -136,6 +136,44 @@ namespace Utilities
   pack_integers(const std::array<std::uint64_t, dim> &index,
                 const int                             bits_per_dim);
 
+  /**
+   * If the library is configured with ZLIB, then this function compresses the
+   * input string and returns a non-zero terminated string containing the
+   * compressed input.
```

Comment:
What about mentioning that the output is a byte string (using the full character range) and should not be outputted on terminal, or a textfile directly? (vtu internally uses a base64 encoding as used later on :-))

On a related note, would it hurt to add a boolean parameter enabling base64 output? That would make this function a bit more useful to use (outside of grid_in)? And it is a simple refactoring :-)

Reply:
I've opened PR #9787 for the base64 part.